projects/ngneat/transloco-persist-lang/src/lib/persist-lang.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Inject, Injectable } from '@angular/core';
+import { isNotBrowser, TranslocoService } from '@ngneat/transloco';
+import { skip } from 'rxjs/operators';
+import { defaults, PersistLangConfig, TRANSLOCO_PERSIST_LANG_CONFIG } from './persist-lang.config';
+import { cookiesStorage } from './cookie-storage';
+import { noopStorage } from './noop-storage';
+
+@Injectable({ providedIn: 'root' })
+export class TranslocoPersistLangService {
+  private mergedConfig: PersistLangConfig;
+
+  constructor(
+    private service: TranslocoService,
+    @Inject(TRANSLOCO_PERSIST_LANG_CONFIG) private config: PersistLangConfig
+  ) {
+    this.mergedConfig = { ...defaults, ...(this.config || {}) };
+    if (isNotBrowser() === false) {
+      this.updateStorageOnLangChange();
+      this.mergedConfig.strategy === 'auto' && this.setActiveLang();
+    }
+  }
+
+  getCachedLang(): string | null {
+    if (isNotBrowser() === false) {
+      return this.getStorage().getItem(this.storageKey);
+    }
+
+    return null;
+  }
+
+  clear() {
+    this.getStorage().removeItem(this.storageKey);
+  }
+
+  private get storageKey() {
+    return this.mergedConfig.storageKey;
+  }
+
+  private updateStorageOnLangChange() {
+    this.service.langChanges$.pipe(skip(1)).subscribe(lang => {
+      this.save(lang);
+    });
+  }
+
+  private setActiveLang() {
+    const cached = this.getStorage().getItem(this.storageKey);
+    const browserLang = this.service.getBrowserLang();
+    const defaultLang = this.service.config.defaultLang;
+    const currentLang = cached || browserLang || defaultLang;
```

Comment:
There are multiple issues with this method.

I understand that this method will only be called if the strategy is `auto`, which is great. However, I think it's implementation will result in trouble.

The first issue is that the library makes a huge assumption on behalf of the developer that the lang format is short - something like *en* rather than *en-US*. One application I worked on used long formats. If the lib makes this assumption, the `currentLang` will end up in the cache as `en`, which will most likely result in an HTTP request for en.json file instead of `en-US.json`, which will return 404.

The second issue is the order of determining the `currengLang`. For example, if a developer provides a `defaultLang` configuration value, it might come as a surprise for the developer to see that this value is not being picked up if there is nothing in the cache. The `browserLang` would get picked up leaving the `defaultLang` kind of useless.

If you'd like to leave the `auto` strategy in place, at least change the order to `const currentLang = cached || defaultLang || browserLang;`
However, I'm wondering if it'd be better to leave it up to the consumer application to implement proper logic of figuring out the current language.
Another way would be to add a new property into the TRANSLOCO_PERSIST_LANG_CONFIG: something like:
```
getLang: (cachedLang: string, defaultLang: string, translocoService: TranslocoService) => string;
```
With this, the developer will be able to specify the logic for determining the language in the consumer app.